src/net/network_manager.cpp

```diff
@@ -28,12 +30,38 @@ void ServerNetworkManager::connect() {
 }
 
 void ServerNetworkManager::register_listeners() {
+    // Menu phase.
+    events::menu::respond_join_event.connect([&](int conn_id, proto::JoinResponse& resp) {
+        proto::ServerMessage msg;
+        msg.set_allocated_join_response(new proto::JoinResponse(resp));
+        server.send_to(conn_id, msg);
+    });
+
     // Build phase.
     events::build::respond_build_event.connect([&](proto::Construct& c) {
         proto::ServerMessage msg;
         msg.set_allocated_build_update(new proto::Construct(c));
         server.send_to_all(msg);
     });
+
+    // Dungeon phase.
+    events::dungeon::network_positions_event.connect([&](std::map<int, Player*> & players) {
+        proto::ServerMessage msg;
+        proto::GameState* state = new proto::GameState();
+        for (auto & pair : players) {
```

Comment:
This is a good point -- doing so would make network manager just be in charge of sending off server messages, but it does push proto creation onto the dispatchers.  I'll look into this after this PR.